### PR TITLE
feat(app): add display sleep settings screen

### PR DESCRIPTION
**Pull request by koji**
# Overview
Currently only the UI. The functionality will be add when we decide how to store data for settings.
Screenshot 2023-02-06 at 2 36 00 PM

# Test Plan
The initial setting value is `Never`.
When tapping one of buttons, the selected button's background will be blue and font color will be white. The previous selected button's background will be gray and the font color will be black.

# Changelog
- add DisplaySleepSettings screen

# Review requests
- the name of the new config and its structure
- create a const for `60 * 1000 * 60 * 24 * 7` in app (App/constants.ts), but seems that app-shell and app-shell-odd don't have a .ts file for constants. Is it okay to hard-code the value?

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] merge edge into this PR when https://github.com/Opentrons/opentrons/pull/12098 is merged into edge

**Comment by koji** [comment]
- [x] add `react-use`
- [x] add a new config
- [x] merge edge and resolve conflicts

**Comment by koji** [comment]
- [x] remove `useEffect` and add useIdle, custom hook
- [x] remove `react-use`
- [x] merge https://github.com/Opentrons/opentrons/pull/12177

**Comment by koji** [comment]
- [x] this branch needs to merge edge after https://github.com/Opentrons/opentrons/pull/12170 is merged into edge
#12170 adds configV14 so this PR will need to update the config version from v14 to v15

File: app/src/App/OnDeviceDisplayApp.tsx [hunk 1]
```diff
@@ -156,31 +161,49 @@ export const onDeviceDisplayRoutes: RouteProps[] = [
   },
 ]
 
+const onDeviceDisplayEvents: Array<keyof DocumentEventMap> = [
```
Comment (ewagoner) [inline app/src/App/OnDeviceDisplayApp.tsx:164]: Are these the only three events that can reset the timer?

**Review by ewagoner — APPROVED** [review]

File: app/src/App/OnDeviceDisplayApp.tsx (hunk 1 above)
Comment (koji) [reply app/src/App/OnDeviceDisplayApp.tsx:164]: Yes, the ODD is a little bit different from normal touch screens.
Initially, I tried events that I put in `useIdle`, but that didn't work as expected and I did try and error to narrow down events. Eventually, these 3 events work with the ODD.

**Pull request merged**